x/programs/rust/examples/counter/src/lib.rs

```diff
@@ -284,5 +237,66 @@ mod tests {
                 .filter_map(|resp| resp.error.as_ref())
                 .next()
         );
+
+        let value = bytes_to_u64(
+            simulator
+                .run_step(
+                    &owner_key,
+                    Step {
+                        endpoint: Endpoint::ReadOnly,
+                        method: "get_value".into(),
+                        max_units: 0,
+                        params: vec![counter2_id.into(), bob_key.clone()],
+                    },
+                )
+                .unwrap()
+                .result
+                .response,
+        );
+        assert_eq!(value, 0);
+
+        simulator
+            .run_step(
+                &owner_key,
+                Step {
+                    endpoint: Endpoint::Execute,
+                    method: "inc_external".into(),
+                    max_units: 100000000,
+                    params: vec![
+                        counter1_id.into(),
+                        counter2_id.into(),
+                        1000000.into(),
+                        bob_key.clone(),
+                        10.into(),
+                    ],
+                },
+            )
+            .unwrap();
+
+        let value = bytes_to_u64(
+            simulator
+                .run_step(
+                    &owner_key,
+                    Step {
+                        endpoint: Endpoint::ReadOnly,
+                        method: "get_value_external".into(),
+                        max_units: 0,
+                        params: vec![
+                            counter1_id.into(),
+                            counter2_id.into(),
+                            1000000.into(),
+                            bob_key.clone(),
+                        ],
+                    },
+                )
+                .unwrap()
+                .result
+                .response,
+        );
+        assert_eq!(value, 10);
+    }
+
+    fn bytes_to_u64(bytes: Vec<u8>) -> u64 {
+        u64::from_le_bytes(bytes.try_into().unwrap())
```

Comment:
Is this a borsh-serialized `u64`? If so, you shouldn't be deserializing it like this, you should be using `borsh::from_slice`

Reply:
True, as per your above comment I will just make the `PlanResponse` accept a generic!